lib/Date.php

```diff
@@ -266,4 +266,36 @@ private function numberOfDaysInMonth(int $year, int $month): int
             \sprintf('%d-%02d-%02d', $year, $month, 1),
         )->format('t');
     }
+
+    /**
+     * @return self The latest date from the provided dates.
+     */
+    public static function max(self $date, self ...$otherDates): self
+    {
+        $maxDate = $date;
+
+        foreach ($otherDates as $otherDate) {
+            if ($otherDate->isAfter($maxDate)) {
+                $maxDate = $otherDate;
+            }
+        }
+
+        return $maxDate;
+    }
+
+    /**
+     * @return self The earliest date from the provided dates.
+     */
+    public static function min(self $date, self ...$otherDates): self
```

Comment:
```suggestion
    public static function min(self $cutOffDate, self ...$otherDates): self
```
So it is clear that the intention here is to use the date in a particular way.